ssl/rio/poll_immediate.c

```diff
@@ -116,7 +392,68 @@ int SSL_poll(SSL_POLL_ITEM *items,
         item->revents = revents;
     }
 
-    /* TODO(QUIC POLLING): Blocking mode */
+out:
+    if (p_result_count != NULL)
+        *p_result_count = result_count;
+
+    return ok;
+}
+
+int SSL_poll(SSL_POLL_ITEM *items,
+             size_t num_items,
+             size_t stride,
+             const struct timeval *timeout,
+             uint64_t flags,
+             size_t *p_result_count)
+{
+    int ok = 1;
+    size_t result_count = 0;
+    ossl_unused int do_tick = ((flags & SSL_POLL_FLAG_NO_HANDLE_EVENTS) == 0);
+    OSSL_TIME deadline;
+
+    /* Trivial case. */
+    if (num_items == 0)
```

Comment:
Shouldn't this still wait for the timeout now that blocking usage is supported?

Reply:
That really depends. It is not too useful call anyway. What do @nhorman @mattcaswell think about this corner case?

Reply:
What does real "poll" do in this case? We should probably mirror it.

Reply:
I think it will sleep. select() will sleep. I've added call to OSSL_sleep().

Reply:
what about `do_tick` here? We are supposed to call to `ossl_quic_reactor_tick()` down in `ossl_quic_conn_poll_events()` but if there are no descriptors to poll here the reactor tick does not happen we just sleep here.

Reply:
We cannot tick if we do not have any SSL objects to tick.